core/lib/zksync_core/src/sync_layer/batch_status_updater/mod.rs

```diff
@@ -0,0 +1,451 @@
+        Ok(Self {
+            last_executed_l1_batch,
+            last_proven_l1_batch,
+            last_committed_l1_batch,
+        })
+    }
+
+    fn extract_tx_hash_and_timestamp(
+        batch_info: &api::BlockDetails,
+        stage: AggregatedActionType,
+    ) -> (Option<H256>, Option<DateTime<Utc>>) {
+        match stage {
+            AggregatedActionType::Commit => {
+                (batch_info.base.commit_tx_hash, batch_info.base.committed_at)
+            }
+            AggregatedActionType::PublishProofOnchain => {
+                (batch_info.base.prove_tx_hash, batch_info.base.proven_at)
+            }
+            AggregatedActionType::Execute => {
+                (batch_info.base.execute_tx_hash, batch_info.base.executed_at)
+            }
+        }
+    }
+
+    fn update(
+        &mut self,
+        status_changes: &mut StatusChanges,
+        batch_info: &api::BlockDetails,
+        stage: AggregatedActionType,
+    ) -> anyhow::Result<()> {
+        let (l1_tx_hash, happened_at) = Self::extract_tx_hash_and_timestamp(batch_info, stage);
+        let (last_l1_batch, updated_changes) = match stage {
+            AggregatedActionType::Commit => (
+                &mut self.last_committed_l1_batch,
+                &mut status_changes.commit,
+            ),
+            AggregatedActionType::PublishProofOnchain => {
+                (&mut self.last_proven_l1_batch, &mut status_changes.prove)
+            }
+            AggregatedActionType::Execute => (
+                &mut self.last_executed_l1_batch,
+                &mut status_changes.execute,
+            ),
+        };
+
+        // Check whether we have all data for the update.
+        let Some(l1_tx_hash) = l1_tx_hash else {
+            return Ok(());
+        };
+        if batch_info.l1_batch_number != last_l1_batch.next() {
+            return Ok(());
+        }
+
+        let action_str = l1_batch_stage_to_action_str(stage);
+        let happened_at = happened_at.with_context(|| {
+            format!("Malformed API response: batch is {action_str}, but has no relevant timestamp")
+        })?;
+        updated_changes.push(BatchStatusChange {
+            number: batch_info.l1_batch_number,
+            l1_tx_hash,
+            happened_at,
+        });
+        tracing::info!("Batch {}: {action_str}", batch_info.l1_batch_number);
+        FETCHER_METRICS.l1_batch[&stage.into()].set(batch_info.l1_batch_number.0.into());
+        *last_l1_batch += 1;
+        Ok(())
+    }
+}
+
+/// Component responsible for fetching the batch status changes, i.e. one that monitors whether the
+/// locally applied batch was committed, proven or executed on L1.
+///
+/// In essence, it keeps track of the last batch number per status, and periodically polls the main
+/// node on these batches in order to see whether the status has changed. If some changes were picked up,
+/// the module updates the database to mirror the state observable from the main node.
```

Comment:
If you are already modifying this component, I'd be for extending this comment to include more information why this component exists in the first place, currently this comment states "updates the database to mirror the state observable from the main node" but for me this is very vague